flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/BaseObserver.java

```diff
@@ -62,9 +62,22 @@ protected void observeJmDeployment(
         JobManagerDeploymentStatus previousJmStatus =
                 deploymentStatus.getJobManagerDeploymentStatus();
 
+        if (JobManagerDeploymentStatus.READY == previousJmStatus) {
+            if (flinkService.isJobManagerServing(effectiveConfig)) {
```

Comment:
Since we are already listing jobs when the deployment is ready, do we really need this? I would prefer we avoid redundant calls to the REST API. The previous delay was there to not run into listJobs error.

Reply:
For JobObserver, I do agree, it is somehow redundant when deployment is ready. For Job, we first `observeJmDeployment`(1 call of `listJobs()`) and then `observeFlinkJobStatus()`(1 or 2 call of `listJobs()`). We could reduce at least 1 extra rest call if we refactor some codes.
But I think we may still need this check here for SessionObserver. For Session, we will only `observeJmDeployment`. Without this check, we will fall into the deployment check always and in current logic of `main` branch, we would change the JobManagerDeploymentStatus to `DEPLOYED_NOT_READY` as well which is not desired.

Reply:
What we could do @bgeng777 is to pass a flag to the JM observer whether to check Rest API for ready jobs. As you explained this should be false for the jobobserver as that checks the rest api anyways. I think for deployed_not_ready state we need to always check to have a good way of telling when it becomes ready.

Reply:
Or we could just keep the if check before observing in the job observer like it was before :)